services/main/internal/services/property.go

```diff
@@ -185,3 +189,98 @@ func (s *propertyService) GetProperty(
 
 	return property, nil
 }
+
+type UpdatePropertyInput struct {
+	PropertyID  string
+	Name        *string
+	Description *string
+	Images      *[]string
+	Tags        *[]string
+	Latitude    *float64
+	Longitude   *float64
+	Address     *string
+	Country     *string
+	Region      *string
+	City        *string
+	GPSAddress  *string
+}
+
+func (s *propertyService) UpdateProperty(
+	context context.Context,
+	input UpdatePropertyInput,
+) (*models.Property, error) {
+	property, err := s.repo.GetByID(context, repository.GetPropertyQuery{ID: input.PropertyID})
+	if err != nil {
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
+			raven.CaptureError(err, nil)
+		}
+		return nil, err
+	}
+
+	if input.Name != nil {
+		property.Name = *input.Name
+	}
+
+	if input.Description != nil {
+		property.Description = input.Description
+	}
```

Comment:
with this logic it means I can never set Description to null? (even though its nullable)

```suggestion
property.Description = input.Description

```